src/main/java/com/epam/training/auction_backend/model/ZooKeeperBiddingStore.java

```diff
@@ -0,0 +1,151 @@
+        return getHighestBidByPath(String.format(AUCTION_BIDS_PATH, auctionId));
+    }
+
+    private UserBidTransferObject getHighestBidByPath(String auctionPath) {
+        try {
+            long maxBid = getMaxBidByAuctionPath(auctionPath);
+            long bidderId = getBidderIdByPath(auctionPath + "/" + maxBid);
+            UserTransferObject user = usersService.getUserById(bidderId).get();
+            return new UserBidTransferObject(user, maxBid);
+        } catch (KeeperException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private long getMaxBidByAuctionPath(String auctionPath) throws KeeperException, InterruptedException {
+        List<String> bids = getZooKeeper().getChildren(auctionPath, true);
+        return bids.stream().mapToLong(Long::valueOf).max().orElse(0);
+    }
+
+    private long getBidderIdByPath(String bidPath) throws KeeperException, InterruptedException {
+        Stat stat = getZooKeeper().exists(bidPath, true);
+        byte[] serializedBidderId = getZooKeeper().getData(bidPath, true, stat);
+        return (long) SerializationUtils.deserialize(serializedBidderId);
+    }
+
+    @Override
+    public boolean makeBid(long auctionId, long amount, long userId) throws BiddingRaceException {
+        String pathForCurrentAuctionsBid = String.format(AUCTION_BIDS_PATH, auctionId);
+        createNodeIfNotExist(pathForCurrentAuctionsBid);
+        makeBid(pathForCurrentAuctionsBid, amount, userId);
+        return true;
+    }
+
+    private void makeBid(String bidsRoot, long bidValue, long userId) throws BiddingRaceException {
+        String currentBidPath = bidsRoot + "/" + String.valueOf(bidValue);
+        try {
+            Stat stat = getZooKeeper().exists(currentBidPath, true);
+            if (stat != null) {
+                byte[] serializedPreviousBidder = getZooKeeper().getData(currentBidPath, true, stat);
+                long previousBidderId = (long) SerializationUtils.deserialize(serializedPreviousBidder);
+                UserTransferObject otherBidder = usersService.getUserById(previousBidderId).get();
+                UserBidTransferObject highestBid = getHighestBidByPath(bidsRoot);
+                throw new BiddingRaceException(otherBidder, highestBid.getBid());
+            } else {
+                getZooKeeper().create(currentBidPath, SerializationUtils.serialize(userId), OPEN_ACL_UNSAFE, PERSISTENT);
+            }
+        } catch (KeeperException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    private ZooKeeper getZooKeeper() {
+        if (zooKeeper == null) {
+            try {
+                zooKeeper = new ZooKeeper(ZOO_KEEPER_CONNECTION_PATH, ZOO_KEEPER_SESSION_TTL, this);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return zooKeeper;
+    }
+
+    private void createNodeIfNotExist(String fullNodePath) {
```

Comment:
I'm not sure if it's a good idea to name method with as a statement, maybe it's better to delete 'IfNotExist' from the name.